inbm/trtl/factory/compose_test.go

```diff
@@ -11,17 +11,33 @@ var mockedExitStatus = 0
 var mockedStdout string
 var mockedStderr string
 
-func Test_is_username_registry_safe_pass(t *testing.T) {
-     result := is_username_registry_safe("user_name.", "server-name:")
-     assert.Equal(t, true, result)
+func TestReturnNilRegistrySafe(t *testing.T) {
+	assert.Equal(t, isRegistryCredentialsSafe("username", "servername"), nil)}
+
+func TestReturnErrorUserNameTooLong(t *testing.T) {
+	expected := errors.New("error: docker registry username can not exceed 30 characters")
+	actual := isRegistryCredentialsSafe("usernameusernameusernameusername", "servername")
+	if actual.Error() != expected.Error() {
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Maintainability$</h3>

In your tests, instead of comparing error strings, consider using `errors.Is` for a more robust error comparison. This approach is less fragile and more idiomatic.